### PR TITLE
Some general minor code quality improvements

### PR DESCRIPTION
**Pull request by unp1**
The issues were:

* these unnecessary double declarations like
for (T x: y) {
var z = x;
....
}
fixed manually.
* reducing complexity of some very few if statements by switching the then-and-else branches (fixed automatically, but automatic fix triggered for each case manually)
* declared but never thrown exceptions (manual fixes)
* some typos in comments I encountered while fixing (manual fixes)

### CODE REVIEW

**Comment by wadoon** [comment]
@unp1 Could you state if the changes are done semi-automatically, whether it was applied structurally, or how you selected certain classes?

**Comment by unp1** [comment]
> @unp1 Could you state if the changes are done semi-automatically, whether it was applied structurally, or how you selected certain classes? This would help for the review.

Hi,

 the issue detection was done automatically, but targeted by selecting specific analysis. The trigger for this issue was just that I stumbled about unnecessary local variable declarations on a different branch like

The issues were:
* these unnecessary double declarations like 
    for (T x: y) {
        var z = x;
        ....
    }
  fixed manually.
* reducing complexity of some very few if statements by switching the then-and-else branches (fixed automatically, but automatic fix triggered for each case manually)
* declared but never thrown exceptions (manual fixes)
* some typos in comments I encountered while fixing (manual fixes)

Best regards, 
Richard

File: key.core/src/main/java/de/uka/ilkd/key/proof/init/DependencyContractPO.java [hunk 1]
```diff
@@ -304,8 +304,7 @@ public void fillSaveProperties(Properties properties) {
      * @return The instantiated proof obligation.
      * @throws IOException Occurred Exception.
```
Comment (FliegendeWurst) [inline key.core/src/main/java/de/uka/ilkd/key/proof/init/DependencyContractPO.java:305]:
```suggestion
```
Same for the other changes.

Comment (unp1) [reply key.core/src/main/java/de/uka/ilkd/key/proof/init/DependencyContractPO.java]: Thanks for the review! Should be resolved with the new commit (hope I did not overlook one).

File: key.ui/src/main/java/de/uka/ilkd/key/gui/actions/QuickSaveAction.java [hunk 2]
```diff
@@ -54,17 +53,15 @@ public static void quickSave(MainWindow mainWindow) {
...
-                LOGGER.debug(status);
-            } catch (IOException x) {
-                mainWindow.popupWarning(
-                    "Quicksaving file " + filename + " failed:\n" + x.getMessage(),
-                    "Quicksave failed");
-                LOGGER.debug("Quicksaving file {} failed.", filename, x);
-            }
+            new ProofSaver(proof, filename, KeYConstants.INTERNAL_VERSION).save();
+            final String status = "File quicksaved: " + filename;
+            mainWindow.popupWarning(
```
Comment (wadoon) [inline key.ui/src/main/java/de/uka/ilkd/key/gui/actions/QuickSaveAction.java:58]: Removed unnecessary if block

**Review by wadoon — APPROVED** [review]

**Pull request merged**